pages/dao/[symbol]/proposal/components/instructions/Raydium/AddLiquidityToPool.tsx

```diff
@@ -0,0 +1,261 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import React, { useContext, useEffect, useState } from 'react'
+import BigNumber from 'bignumber.js'
+import * as yup from 'yup'
+import { BN } from '@project-serum/anchor'
+import {
+  Governance,
+  ProgramAccount,
+  serializeInstructionToBase64,
+} from '@solana/spl-governance'
+import { PublicKey } from '@solana/web3.js'
+import Input from '@components/inputs/Input'
+import Select from '@components/inputs/Select'
+import useGovernedMultiTypeAccounts from '@hooks/useGovernedMultiTypeAccounts'
+import useRealm from '@hooks/useRealm'
+import { createAddLiquidityInstruction } from '@tools/sdk/raydium/createAddLiquidityInstruction'
+import {
+  getAmountOut,
+  getLiquidityPoolKeysByLabel,
+} from '@tools/sdk/raydium/helpers'
+import { liquidityPoolKeysList } from '@tools/sdk/raydium/poolKeys'
+import { debounce } from '@utils/debounce'
+import { isFormValid } from '@utils/formValidation'
+import {
+  AddLiquidityRaydiumForm,
+  UiInstruction,
+} from '@utils/uiTypes/proposalCreationTypes'
+import useWalletStore from 'stores/useWalletStore'
+
+import { NewProposalContext } from '../../../new'
+import GovernedAccountSelect from '../../GovernedAccountSelect'
+
+const AddLiquidityToPool = ({
+  index,
+  governance,
+}: {
+  index: number
+  governance: ProgramAccount<Governance> | null
+}) => {
+  const connection = useWalletStore((s) => s.connection)
+  const wallet = useWalletStore((s) => s.current)
+  const { realmInfo } = useRealm()
+  const { governedMultiTypeAccounts } = useGovernedMultiTypeAccounts()
```

Comment:
There is new function `getGovernedAccountPublicKey`:

```
  const {
    governedMultiTypeAccounts,
    getGovernedAccountPublicKey,
  } = useGovernedMultiTypeAccounts()
```

```
    const pubkey = getGovernedAccountPublicKey(form.governedAccount, true)

    if (!pubkey) return invalid
```

Even though this method is still to be improved because it's not clear yet if we target the SOL treasury account itself or it's owner. Available to discuss it with you and find an appropriate solution

https://github.com/UXDProtocol/governance-ui/blob/719abd087768bf26637341082ae389b6edb8d2b8/pages/dao/%5Bsymbol%5D/proposal/components/instructions/socean/MintBondedTokens.tsx